doc: update documentation with default login details for MetalK8s UI & Grafana service (#2377)
scality/metalk8s: docs/operation/account_administration.rst

======================

This section highlights **MetalK8s Account Administration** which covers
user authentication, identity management and access control.

User Authentication and Identity management
-------------------------------------------

Identity management and user authentication in MetalK8s is driven by the
integration of `kube-apiserver` and Dex (an OIDC provider).

> While `kube-apiserver` in a MetalK8s cluster enables OpenID Connect(OIDC) as an
> authentication strategy, Dex is configured to provide a static user store and

Reviewer: Not sure about the contrast you're implying with `While ...`. Maybe:

```suggestion
Kubernetes API enables OpenID Connect (OIDC) as one authentication strategy
(it also supports certificate-based authentication) by trusting Dex as an
OIDC Provider.

Dex can authenticate users against:

 - a static user store (stored in configuration)
 - a connector-based interface, allowing to plug in external ...
```

> a connector based interface to allow MetalK8s users plug-in sub-systems like
> LDAP, SAML, GitHub, Active Directory and others.

Reviewer: ```suggestion
a connector-based interface which allows MetalK8s users to plug in external
identity providers such as LDAP, SAML, GitHub, Active Directory and others.
```

MetalK8s OIDC based Services
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> MetalK8s out of the box enables OpenID Connect(OIDC) based authentication for

Reviewer: Parentheses!! 
```suggestion
MetalK8s out of the box enables OpenID Connect (OIDC) based authentication for
```

its UI and Grafana service.

.. _ops-grafana-admin:

Administering Grafana and MetalK8s UI
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

A fresh installation of MetalK8s has its UI and Grafana service with default
login credentials as: ``admin@metalk8s.invalid`` / ``password``.

> The default credentials are defined in the OIDC provider static user store to

Reviewer: ```suggestion
This default user is defined in Dex configuration as a static user,
to allow...
```

allow MetalK8s administrators first time access to these services. It is
recommended that MetalK8s administrators change the default password.

.. note::

>    The MetalK8s UI and the Grafana service instance make use of the same local

Reviewer 2: maybe ask @lucieleonard to review the sentence

>    The MetalK8s UI and the Grafana service instance make use of the same local
>    user store and so changing the default credentials, has a direct impact on
>    both services.

Reviewer: Maybe:
```suggestion
   The MetalK8s UI and Grafana are both configured to use OIDC as
   an authentication mechanism, and trust Dex as a Provider. Changing
   the Dex configuration, including the default credentials, will impact
   both UIs.
```

.. _ops-grafana-admin:

> Administering Grafana and MetalK8s GUI

Reviewer: I think this doc needs to be rewritten, to show the following sections:

- Dex and OIDC, how it works (connectors, static users...)
- Services relying on Dex (K8s API, Salt API, Grafana, MetalK8s GUI...)
- Procedures to add/edit static users
- Procedures to configure connectors (maybe not filled in right now)

Or, if we really want to put everything under a single "Cluster and Services Configurations" document (I'd vote against this, harder to discover), then maybe just remove this document entirely.

Author: Done!
